test/framework/kubetest/setup.go

```diff
@@ -0,0 +1,106 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubetest
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	ciVersionURL = "https://dl.k8s.io/ci/k8s-master.txt"
+	// E2EVariableUseCIArtifacts if "true" will fetch the latest build from the main branch of kubernetes/kubernetes
+	E2EVariableUseCIArtifacts = "KUBETEST_USE_CI_ARTIFACTS"
+	// E2EVariableConfigFilePath is the path to the kubetest e2e config file
+	E2EVariableConfigFilePath = "KUBETEST_CONFIG_FILE_PATH"
+	// E2EVariableGinkgoNodes is the number of Ginkgo nodes to use
+	E2EVariableGinkgoNodes = "KUBETEST_GINKGO_NODES"
+	// E2EVariableGinkgoSlowSpecThreshold is time in s before spec is marked as slow
+	E2EVariableGinkgoSlowSpecThreshold = "KUBETEST_GINKGO_SLOW_SPEC_THRESHOLD"
+)
+
+// Configuration creates a new kubetest configuration
+type Configuration struct {
```

Comment:
The type `Configuration` confused me.
As far as I understand the code, this struct does not define the kubetest configuration, but it carries info from `NewConfiguration` to `Run` where the actual works happen.
Also, among the `Configuration`  informations there are two nested
- `E2EConfig` config (which basically carries variables)
- `ConfigFilePath` that points to the real kubetest configuration

What about getting rid of this struct, and implement the `Run` func getting all the required info in input via `RunInput`? this will avoid confusion an make this code more similar to other test.Framework helpers..

Also, is it possible to pass the required values directly, without having a direct dependency on E2EConfig?